src/hotspot/share/classfile/classLoaderHierarchyDCmd.cpp

```diff
@@ -154,33 +154,59 @@ class LoaderTreeNode : public ResourceObj {
   // one.
   int _num_folded;
 
+  // Returns Klass of loader; NULL for bootstrap loader
+  const Klass* loader_klass() const {
+    return (_loader_oop != NULL) ? _loader_oop->klass() : NULL;
+  }
+
+  // Returns ResourceArea-allocated class name of loader class
+  const char* loader_class_name() const {
+    const Klass* klass = loader_klass();
+    return klass != NULL ? klass->external_name() : NULL;
+  }
+
+  // Returns oop of loader name; NULL for bootstrap; NULL if no name was set
+  oop loader_name_oop() const {
+    return (_loader_oop != NULL) ? java_lang_ClassLoader::name(_loader_oop) : NULL;
+  }
+
+  // Returns ResourceArea-allocated name of loader, NULL if none is set
+  const char* loader_name() const {
+    oop name_oop = loader_name_oop();
+    return name_oop != NULL ? java_lang_String::as_utf8_string(name_oop) : NULL;
+  }
+
+  bool is_bootstrap() const {
+    if (_loader_oop == NULL) {
+      assert(_cld != NULL && _cld->is_boot_class_loader_data(), "bootstrap loader must have CLD");
+      return true;
+    }
+    return false;
+  }
+
   void print_with_childs(outputStream* st, BranchTracker& branchtracker,
```

Comment:
I know this isn't part of your changes, but the plural of child is children, not childs. However, I think child_nodes would probably be best here.

Reply:
Fixed

Reply:
It looks like there are a couple more references to "childs" still present.